src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentValidator.java

```diff
@@ -0,0 +1,89 @@
+        } catch (IOException e) {
+            LOG.error("Error while getting assignment from canvas for section: " + attendanceAssignment.getAttendanceSection().getSectionId(), e);
+            return new Error("Could not connect to Canvas to get assignment");
+        } catch (ObjectNotFoundException e) {
+            LOG.debug("Assignment with ID: " + attendanceAssignment.getCanvasAssignmentId() + "not found in Canvas.");
+            return new Error("NO CANVAS ASSIGNMENT LINKED");
+        }
+
+        if(!assignmentOptional.isPresent()) {
+            return new Error("NO CANVAS ASSIGNMENT LINKED");
+        }
+
+        return null;
+    }
+
+    public Error validateCanvasAssignment(CourseConfigurationForm courseConfigurationForm, Long courseId, AttendanceAssignment attendanceAssignment, CanvasApiWrapperService canvasApiWrapperService, OauthToken oauthToken) {
+
+        //Look for changes between the course configuration in the form and in the DB
+        if(!compareFormToDB(courseConfigurationForm, attendanceAssignment)) {
+            LOG.debug("Configuration form is different than saved assignment configuration for section: " + attendanceAssignment.getAttendanceSection().getSectionId());
+            return new Error("Assignment configuration needs to be saved before pushing to Canvas");
+        }
+
+        //Look for the canvas assignment in canvas
+        Optional<Assignment> assignmentOptional;
+        try {
+            assignmentOptional = canvasApiWrapperService.getSingleAssignment(courseId, oauthToken, attendanceAssignment.getCanvasAssignmentId().toString());
+        } catch (IOException e) {
+            LOG.error("Error while getting assignment from canvas for section: " + attendanceAssignment.getAttendanceSection().getSectionId(), e);
+            return new Error("Could not connect to Canvas to get assignment");
+        }
+
+        if(!assignmentOptional.isPresent()) {
+            return new Error("Assignment not found in Canvas");
+        }
+
+        //Looks for discrepancy between the assignment in canvas and in database
+        if (assignmentOptional.get().getPointsPossible().doubleValue() != courseConfigurationForm.getAssignmentPoints().doubleValue()) {
+            LOG.debug("Discrepancy between Canvas and DB assignment. Point value of Canvas assignment is: " + assignmentOptional.get().getPointsPossible().doubleValue() +
+                      " and point value of Database assignment is :" + courseConfigurationForm.getAssignmentPoints().doubleValue());
+            return new Error("DISCREPANCY BETWEEN CANVAS AND DATABASE");
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns true if configuration in the form and in the db is the same
+     */
+    private boolean compareFormToDB(CourseConfigurationForm courseConfigurationForm, AttendanceAssignment attendanceAssignment) {
```

Comment:
Could the name of this method reflect the return value somehow? When the method is used above, it isn't obvious when the compare returns true or false by the name alone.